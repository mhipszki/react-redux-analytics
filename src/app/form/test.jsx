import React from 'react';
import { shallow } from 'enzyme';

import { Form } from './';

test('renders a text input and a button', () => {
  const component = shallow(<Form />);
  expect(component).toMatchSnapshot();
});

test('runs callback when text input changes', () => {
  const onTextChange = jest.fn();
  const component = shallow(<Form onTextChange={onTextChange} />);
  const textInput = component.find('text-input__TextInput');
  const event = { type: 'change' };
  textInput.simulate('change', event);
  expect(onTextChange).toHaveBeenCalledWith(event);
});

test('runs callback when message is sent', () => {
  const onButtonClick = jest.fn();
  const component = shallow(<Form onButtonClick={onButtonClick} />);
  const button = component.find('button__Button');
  const event = { type: 'click' };
  button.simulate('click', event);
  expect(onButtonClick).toHaveBeenCalledWith(event);
});

test('disables text input and button while message is being sent', () => {
  const component = shallow(<Form isSending />);
  const textInput = component.find('text-input__TextInput');
  const button = component.find('button__Button');
  expect(textInput.props().disabled).toBe(true);
  expect(button.props().disabled).toBe(true);
});

test('notifies user that message is being sent', () => {
  const component = shallow(<Form isSending />);
  const heading = component.find('heading__Heading');
  expect(heading.render().text()).toBe('Sending message...');
});

test('notifies user that message has been delivered', () => {
  const component = shallow(<Form sentIn={1234} />);
  const heading = component.find('heading__Heading');
  expect(heading.render().text()).toBe('Message was sent in 1234ms');
});
