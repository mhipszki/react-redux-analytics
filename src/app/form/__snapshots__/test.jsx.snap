// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text input and a button 1`] = `
<column__Column>
  <heading__Heading>
    React / Redux / Analytics demo
  </heading__Heading>
  <line>
    Demo time!
  </line>
  <text-input__TextInput
    placeholder="type your message here"
  />
  <button__Button>
    Send message
  </button__Button>
</column__Column>
`;
